README.md

```diff
@@ -1,5 +1,146 @@
 # aedes-server-factory
 
-Aedes server factory. Supports tcp, tls, http, https, http2, ws, wss and proxy decoders
+Aedes server factory. Supports tcp, http, http2, ws, and proxy decoders
 
-Work In Progress
\ No newline at end of file
+Work In Progress: tls, https, http2, wss
+
+## Install
+
+To install aedes-server-factory :
+
+```sh
+npm install aedes-server-factory
+```
+
+# Options
+
+## ws
+
+boolean used to wrap a http | http2 | https server in a websocket server.
+
+`createServer` will return the http server if not http options is defined.
+
+## http2
+
+boolean used to create http2 server.
+
+## https
+
+boolean used to create http | http2 secure server.
+
+## tls
+
+boolean used to create tls server.
```

Comment:
http2, tls, https, ws... I do not think they can be booleans. They'll all have custom options that could be set.

Reply:
Ok, the options are updated to take the type of the corresponding option of the server to create